test/test_arraycontext.py

```diff
@@ -120,6 +120,14 @@ def __init__(self, actx, data):
 
     __array_priority__ = 10
 
+    def __bool__(self):
+        if len(self) == 1 and len(self.data[0]) == 1:
```

Comment:
I think this checks whether the only group array is 1-long along its 0th-axis, but we might want to check if has a size of 1.

Reply:
Good catch! Updated in 78dfbe6.

I added `__bool__` to meshmode in inducer/meshmode#251 too, but without the check for scalars. That forces a `to_numpy` essentially, so it seemed like a good idea to avoid it all together and just use this `any` / `all`. What do you think?

Reply:
Yep, I too am in favor of avoiding implicit `to_numpy`s.